### ISSUE TITLE
Interceptor exclusion behavior is not expected

### ISSUE THREAD

**Issue by mamun217** [grails/grails-core#10857]
It seem to me that the interceptor exclusion pattern is not working correctly.
I have an interceptor with before() event to authorize every request except the management endpoints. 

Interceptor exclusion is configured as below:
``` 
    AuthorizationInterceptor() {
        matchAll().excludes(uri: "/mgmt/*")
    }

    boolean before() {
        // authorization logic. If authorized then returns true, otherwise false
    }
```

Management endpoint uses /mgmt context path.
```
    endpoints:
        enabled: true
    management:
        context-path: "/mgmt"
```

### Steps to Reproduce

1. Start the application with above configuration
2. Enter /mgmt/health from the browser

### Expected Behaviour

Interceptor should exclude /mgmt/health request

### Actual Behaviour

Interceptor doesn't exclude /mgmt/health and try to authorize using the code written under `before()` method of the interceptor.

**Note:** When application is running without the interceptor, accessing /mgmt/health responds as expected.
Accessing this URL `http://localhost:8080/APP-STAGE/mgmt/health`
Returns `{"status":"UP"}`

### Environment Information

- **Operating System**: Windows 7 Enterprise
- **Grails Version:** 3.3.0
- **JDK Version:** 1.8.0_74

**Comment by ilopmar** [comment]
I've just tested with Grails 3.3.2 and it is working as expected. The interceptor is not executed for the `/mgmt` endpoint.

I've created a sample project that shows it's working: https://github.com/grails-core-issues-forks/grails-issue-10857. 

Can you please check it?

**Comment by mamun217** [comment]
I probably haven't clearly mentioned one important part of the issue, **context path**. The application has to start under a given context path. Not able to make exclusion work for that scenario. Log from the before method prints like below. Controller and action names print as **null**. 
`interceptor.AuthorizationInterceptor     : --- Interceptor BEFORE --- Controller=null, Action=null`

In **Actual Behavior** section I have given a sample URL (`http://localhost:8080/APP-STAGE/mgmt/health`). One more thing is that we are using **Grails 3.3.0**.

**Comment by ilopmar** [comment]
@mamun217 you were right. Setting the application context reproduces the bug. I've submitted this PR #10912 with the fix, but I'm afraid that the fix won't be for Grails 3.3.0 but for Grails 3.3.3.

**Comment by ilopmar** [comment]
Fixed by #10912

**Comment by mamun217** [comment]
@ilopmar Thanks. For now, I am excluding it manually on the before() method.
